kernel/fs.c

```diff
@@ -336,14 +340,57 @@ int fs_file_write(struct fs_file *file, const char *buffer, uint32_t length, uin
 	return total;
 }
 
-int fs_file_get_dimensions(struct fs_file *f, int * dims, int n) 
+int fs_dirent_dup(struct fs_dirent *src, struct fs_dirent *dst) {
+	char *buffer = kmalloc(4096);
```

Comment:
Yikes, magic numbers!   At least use the macro PAGE_SIZE, or even better, get the blocksize from the underlying file system.